packages/ui/src/elements/Thumbnail/createThumbnail.ts

```diff
@@ -1,50 +1,94 @@
 /**
  * Create a thumbnail from a File object by drawing it onto an OffscreenCanvas
  */
-export const createThumbnail = (file: File): Promise<string> => {
+export const createThumbnail = (
+  file: File,
+  fileSrc?: string,
+  mimeType?: string,
+): Promise<string> => {
   return new Promise((resolve, reject) => {
-    const img = new Image()
-    img.src = URL.createObjectURL(file) // Use Object URL directly
-
-    img.onload = () => {
-      const maxDimension = 280
-      let drawHeight: number, drawWidth: number
-
-      // Calculate aspect ratio
-      const aspectRatio = img.width / img.height
-
-      // Determine dimensions to fit within maxDimension while maintaining aspect ratio
-      if (aspectRatio > 1) {
-        // Image is wider than tall
-        drawWidth = maxDimension
-        drawHeight = maxDimension / aspectRatio
-      } else {
-        // Image is taller than wide, or square
-        drawWidth = maxDimension * aspectRatio
-        drawHeight = maxDimension
-      }
+    /**
+     * Create DOM element
+     * Draw media on offscreen canvas
+     * Resolve fn promise with the base64 image url
+     * @param media
+     * @param maxDimension
+     */
+    const _getBase64ImageUrl = async (
+      media: HTMLImageElement | HTMLVideoElement,
+      maxDimension = 280,
+    ): Promise<string> => {
+      return new Promise((_resolve, _reject) => {
+        let drawHeight: number, drawWidth: number
+
+        // Calculate aspect ratio
+        const width = media.width || (media as HTMLVideoElement).videoWidth
+        const height = media.height || (media as HTMLVideoElement).videoHeight
+        const aspectRatio = width / height
+
+        // Determine dimensions to fit within maxDimension while maintaining aspect ratio
+        if (aspectRatio > 1) {
+          // Image is wider than tall
+          drawWidth = maxDimension
+          drawHeight = maxDimension / aspectRatio
+        } else {
+          // Image is taller than wide, or square
+          drawWidth = maxDimension * aspectRatio
+          drawHeight = maxDimension
+        }
+
+        // Create an OffscreenCanvas
+        const canvas = new OffscreenCanvas(drawWidth, drawHeight)
+        const ctx = canvas.getContext('2d')
+
+        // Draw the image onto the OffscreenCanvas with calculated dimensions
+        ctx.drawImage(media, 0, 0, drawWidth, drawHeight)
+
+        // Convert the OffscreenCanvas to a Blob and free up memory
+        canvas
+          .convertToBlob({ type: 'image/jpeg', quality: 0.25 })
```

Comment:
type `image/jpeg` instead of `image/png` could display a full back thumbnail in case the image has a transparent background like this 

<img width="258" alt="Screenshot 2024-11-03 at 14 55 58" src="https://github.com/user-attachments/assets/4ed7b5ad-634e-4cab-a5fd-2792695f3aad">
<img width="281" alt="Screenshot 2024-11-03 at 14 56 08" src="https://github.com/user-attachments/assets/022c4bac-bd8b-4cd5-99bb-5873f776a484">